atc/resource/fetcher.go

```diff
@@ -71,6 +73,9 @@ func (f *fetcher) Fetch(
 	defer ticker.Stop()
 
 	versionedSource, err := f.fetchWithLock(ctx, logger, source, imageFetchingDelegate.Stdout())
+	if containerLock != nil {
+		containerLock.Release()
+	}
```

Comment:
the lock would need to be released after the second call to `fetchWithLock()` on line 86 since container creation could occur there, but this is susceptible to deadlock because we have 2 locks being held/released here.